### PR TITLE
Support Swift 2.3

### PR DESCRIPTION
**Pull request by ikesyo**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,29 +1,25 @@
...
-      env: JOB=SPM
-      install:
-        - eval "$(curl -sL https://gist.githubusercontent.com/kylef/5c0475ff02b7c7671d2a/raw/02090c7ede5a637b76e6df1710e83cd0bbe7dcdf/swiftenv-install.sh)"
-    - script: swift build
-      env: JOB=Linux
-      sudo: required
-      dist: trusty
-      language: generic
-      install:
-        - eval "$(curl -sL https://gist.githubusercontent.com/kylef/5c0475ff02b7c7671d2a/raw/02090c7ede5a637b76e6df1710e83cd0bbe7dcdf/swiftenv-install.sh)"
```
Comment (mdiep) [inline .travis.yml]: Why did this job need to be removed?

Comment (ikesyo) [inline .travis.yml]: I tried to update to a recent snapshot as of https://github.com/jpsim/Commandant/tree/jp-05-31-snapshot, but that requires breaking changes and I gave it up. So this should be my fault at that time.

Comment (ikesyo) [inline .travis.yml]: I will restore this.

**Comment by ikesyo** [comment]
Oh, I remembered that [Result 2.1.0](https://github.com/antitypical/Result/releases/tag/2.1.0) breaks the builds for DEVELOPMENT-SNAPSHOT-2016-03-24-a.

> Swift 3 support has been updated to the 2016-05-31 snapshot

But updating to DEVELOPMENT-SNAPSHOT-2016-05-31-a requires a large number of `#if swift(>=3)` and I'm not a fun of it. That is the reason I dropped the SwiftPM tests for now.

**Comment by ikesyo** [comment]
Should we support Swift 2.3 and Swift 3 (DEVELOPMENT-SNAPSHOT-2016-05-31-a) simultaneously as Result does?

**Comment by mdiep** [comment]
I think Commandant can stick to supporting just 1 version of Swift.

**Comment by ikesyo** [comment]
Updating `Package.swift` to use Result 2.0.0 fixed the issue. Unfortunately Travis is in trouble 😞 https://www.traviscistatus.com/incidents/rqb906yypnk0.

**Comment by mdiep** [comment]
✨ 

Just waiting on Travis now. 👍

**Pull request merged**